### PR TITLE
Create CONTRIBUTING.md

### LINKED ISSUE #834: Create CONTRIBUTING.md
It would be useful to have CONTRIBUTING.md for this project.

The [Pyro CONTRIBUTING.md](https://github.com/pyro-ppl/pyro/blob/dev/CONTRIBUTING.md) can serve as a nice template here.

If the maintainers of the project are happy with the idea, I can draft a version in a form of a PR.

### PR DESCRIPTION
**Pull request by ab-10**
Closes #834

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+pytest -vs -n auto
+```
+
+To run a single test from the command line
+```sh
+pytest -vs {path_to_test}::{test_name}
+# or in cuda mode
+CUDA_TEST=1 PYRO_TENSOR_TYPE=torch.cuda.DoubleTensor pytest -vs {path_to_test}::{test_name}
+```
+
```
Comment (fehiepsi) [inline CONTRIBUTING.md:38]: Could you add a small Profiling section with a sentence to point to [Profiling JAX programs](https://jax.readthedocs.io/en/latest/profiling.html)?

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+```sh
+pip install pytest-xdist
+pytest -vs -n auto
+```
+
+To run a single test from the command line
+```sh
+pytest -vs {path_to_test}::{test_name}
+# or in cuda mode
+CUDA_TEST=1 PYRO_TENSOR_TYPE=torch.cuda.DoubleTensor pytest -vs {path_to_test}::{test_name}
```
Comment (fehiepsi) [inline CONTRIBUTING.md:36]: This should be
```
# or in cuda mode and double precision
JAX_PLATFORM_NAME=gpu JAX_ENABLE_X64=1 pytest -vs {path_to_test}::{test_name}
```

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+# Submitting
+
+For larger changes, please open an issue for discussion before submitting a pull request.
```
Comment (fehiepsi) [inline CONTRIBUTING.md:41]: This `For relevant design questions to consider, see past [design documents](https://github.com/pyro-ppl/pyro/wiki/Design-Docs).` is also important.

File: CONTRIBUTING.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+# install jax/jaxlib first for CUDA support
+pip install -e .[dev]  # contains additional dependencies for NumPyro development
+```
+
+# Testing
+
+Before submitting a pull request, please autoformat code and ensure that unit tests pass locally
+```sh
+make format            # runs isort
+make test              # linting and unit tests
```
Comment (fehiepsi) [inline CONTRIBUTING.md:23]: could you add `make doctest  # test module’s docstrings` here too

**Review by fehiepsi — COMMENTED** [review]
Thanks a lot, @ab-10! This looks great overall. Could you help me add this [README.md](https://github.com/pyro-ppl/pyro/blob/dev/docs/README.md) file to the docs folder?

**Comment by ab-10** [comment]
Thank you for your prompt review. I hope I've addressed all the comments, but in case anything is subpar please let me know.

File: CONTRIBUTING.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+Please follow our established coding style including variable names, module imports, and function definitions.
+The NumPyro codebase follows the [PEP8 style guide](https://www.python.org/dev/peps/pep-0008/)
+(which you can check with `make lint`) and follows
+[`isort`](https://github.com/timothycrosley/isort) import order (which you can enforce with `make format`).
+
+# Setup
+
+To set up local development environment, install NumPyro from source:
+
+```
```
Comment (fehiepsi) [inline CONTRIBUTING.md:12]: nit: missing ```sh

File: CONTRIBUTING.md [hunk 6]
```diff
@@ -0,0 +1,59 @@
...
+pip install -e .[dev]  # contains additional dependencies for NumPyro development
+```
+
+# Testing
+
+Before submitting a pull request, please autoformat code and ensure that unit tests pass locally
+```sh
+make format            # runs isort
+make test              # linting and unit tests
+make doctest           # test module’s docstrings
```
Comment (fehiepsi) [inline CONTRIBUTING.md:24]: nit: `’` -> `'`

**Review by fehiepsi — APPROVED** [review]
LGTM! Thanks, @ab-10! Could you address the following nits? :)

**Comment by ab-10** [comment]
Done, good spot.

**Review by fehiepsi — APPROVED** [review]

**Pull request merged**